go/multiorch/recovery/analysis/generator.go

```diff
@@ -480,3 +496,85 @@ func (g *AnalysisGenerator) isReplicaConnectedToPrimary(
 
 	return true
 }
+
+// checkTimelineDivergence checks if a replica has a diverged timeline.
+// A timeline is diverged if the replica is on an older timeline and has
+// WAL past the point where the primary's timeline forked.
+func (g *AnalysisGenerator) checkTimelineDivergence(
+	primary *multiorchdatapb.PoolerHealthState,
+	replica *multiorchdatapb.PoolerHealthState,
+) *multiorchdatapb.ShardProblem {
+	// Get timeline info from both nodes
+	var primaryTL *consensusdatapb.TimelineInfo
+	var replicaTL *consensusdatapb.TimelineInfo
+
```

Comment:
Should we have a sanity check that primary and replica are not the same multipooler?